include/triton/core/tritonserver.h

```diff
@@ -569,6 +572,21 @@ typedef void (*TRITONSERVER_InferenceTraceActivityFn_t)(
     TRITONSERVER_InferenceTraceActivity activity, uint64_t timestamp_ns,
     void* userp);
 
+/// Type for trace tensor activity callback function. This callback function
+/// is used to report tensor activity occurring for a trace. This function
+/// does not take ownership of 'trace' and so any information needed
+/// from that object must be copied before returning. The 'userp' data
+/// is the same as what is supplied in the call to
+/// TRITONSERVER_InferenceTraceTensorNew.
+typedef void (*TRITONSERVER_InferenceTraceTensorActivityFn_t)(
+    TRITONSERVER_InferenceTrace* trace,
+    TRITONSERVER_InferenceTraceActivity activity, const std::string& name,
+    TRITONSERVER_DataType datatype, const void* base, size_t byte_size,
+    const int64_t* shape, uint64_t dim_count,
+    TRITONSERVER_MemoryType memory_type, int64_t memory_type_id,
+    const std::string& detail,  // detail information of the tensor
```

Comment:
This is a C API so can't have std:string. Also, we likely want to be more specific then having a generic field. So for now just remove this.